utils/fvwm-version-str.sh

```diff
@@ -10,12 +10,180 @@
 #
 # Intended to be called from configure.ac (via autogen.sh)
 
-VERSION="released"
+# Modified from gitdescribe project (https://github.com/pete4abw/gitdescribe)
+# Peter Hyman, pete@peterhyman.com
+# December 2020
+# Free to use
+# No warranties
+# Attribution appreciated as well as enhancements or fixes!
 
-[ -d ".git" ] || { echo "$VERSION" && exit 0 ; }
+usage() {
+cat >&2  <<EOF
+$(basename $0) command [-r]
+all - entire git describe
+commit - commit, omitting g
+tagrev - tag revision count
+major - major release version
+minor - minor release version
+micro - micro release version - may include commit info if not release tag
+reldatel - Release tag date long Monthname format (DD MMMMMMM YYYY)
+reldates - Release tag date short Monthname format (DD-Mmm-YYYY)
+reldaten - Release tag date number format only (YYYY-m#-DD)
+version - M.m.c
+version_info - released or null
+-r -- get release tag only
+EOF
+exit 1
+}
 
-if grep -q -i '^ISRELEASED="yes"' ./configure.ac; then
-	echo "$VERSION"
-else
-	git describe --always --long --dirty
+# showw message and usage
+die() {
+	echo "$1"
+	usage
+}
+
+# return variables
+# everything, with leading `v' and leading `g' for commits
+describe_tag=
+
+# abbreviated commit
+commit=
+
+# count of commits from last tag
+tagrev=
+
+# major version
+major=
+
+# minor version
+minor=
+
+# micro version
+micro=
+
+# release date long format
+releasedatel=
+
+# release date short format
+releasedates=
+
+# release date number format
+releasedaten=
+
+# version
+version=
+
+# version info, release or null
+version_info=
+
+# get release or tag?
+# if -r option is used, tagopt will be null
+tagopt="--tags"
+
+# how long shhow commit string be? 7 is default
+commit_length=7
+
+# get whole commit and parse
+# describe_tag variable will hold
+# TAG-R-C
+# with leading v and commit leadint g removed
+# tag commit dates obtained with git show
+
+init() {
+	local r_describe
+	# git describe raw format
+	describe_tag=$(git describe $tagopt --long --abbrev=$commit_length)
+
+	# git describe prefixes commit with the letter `g'
+	describe_tag=${describe_tag/g/}
```

Comment:
Ah.  Please don't do that.  The "g-" prefix is useful and I use it all the time to run `git checkout` on it.  Removing this "g" breaks that.  It's actually there for a reason.  From `git-describe(1)`:

```
       The number of additional commits is the number of commits which would be displayed by "git log v1.0.4..parent". The hash suffix is "-g" + unambiguous abbreviation for the tip commit of parent
       (which was 2414721b194453f058079d897d13c4e377f92dc6). The "g" prefix stands for "git" and is used to allow describing the version of a software depending on the SCM the software is managed with.
       This is useful in an environment where people may use different SCMs.
```